lib/mapObjects/CQuest.h

```diff
@@ -17,49 +17,27 @@ VCMI_LIB_NAMESPACE_BEGIN
 
 class CGCreature;
 
-class DLL_LINKAGE CQuest final
+class DLL_LINKAGE CQuest: public Rewardable::Limiter
 {
-	mutable std::unordered_map<ArtifactID, unsigned, ArtifactID::hash> artifactsRequirements; // artifact ID -> required count
-
 public:
-	enum Emission {
-		MISSION_NONE = 0,
-		MISSION_LEVEL = 1,
-		MISSION_PRIMARY_STAT = 2,
-		MISSION_KILL_HERO = 3,
-		MISSION_KILL_CREATURE = 4,
-		MISSION_ART = 5,
-		MISSION_ARMY = 6,
-		MISSION_RESOURCES = 7,
-		MISSION_HERO = 8,
-		MISSION_PLAYER = 9,
-		MISSION_HOTA_MULTI = 10,
-		// end of H3 missions
-		MISSION_KEYMASTER = 100,
-		MISSION_HOTA_HERO_CLASS = 101,
-		MISSION_HOTA_REACH_DATE = 102
-	};
 
-	enum Eprogress {
+	enum EProgress {
 		NOT_ACTIVE,
 		IN_PROGRESS,
 		COMPLETE
 	};
```

Comment:
FYI - this enum does not works as intended.
"in progress" should be tracked per player (e.g. whether player visited corresponding seer hut)
"complete" should be tracked per quest.

Probably should be replaced with `bool isQuestCompleted` and for visitation status use existing `ChangeObjectVisitors` pack.